components-registry-dsl/src/main/kotlin/org/octopusden/octopus/components/registry/dsl/script/ComponentsRegistryScriptRunner.kt

```
@@ -43,19 +41,35 @@ object ComponentsRegistryScriptRunner {
-            KotlinJsr223DefaultScriptEngineFactory().scriptEngine
-        } catch (e: Exception) {
-            logger.info("Unable to get default kotlin script engine, fallback to local script engine")
-            KotlinJsr223JvmLocalScriptEngineFactory().scriptEngine
+            // Always switch to buildscript classloader where Kotlin scripting classes are loaded
+            val properCl = this::class.java.classLoader
+            Thread.currentThread().contextClassLoader = properCl
+
+            KotlinJsr223DefaultScriptEngineFactory().scriptEngine.also {
+                logger.info("Using KotlinJsr223DefaultScriptEngineFactory")
+            }
+        } catch (e: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)